# HG changeset patch
# User roland
# Date 1418632606 -3600
# Node ID a733dad6fc1e2572ed227e898da35e0053cbb7c5
# Parent  db035d4ba1bd25ac8803bb2d177cb35681eb6907
8067231: Zero builds fails after JDK-6898462
Summary: Interpreter::remove_activation_entry() is not defined for the C++ interpreter
Reviewed-by: roland, coleenp
Contributed-by: Severin Gehwolf <sgehwolf@redhat.com>

--- jdk8/hotspot/src/share/vm/interpreter/interpreterRuntime.cpp	Sat Dec 13 01:24:10 2014 +0300
+++ jdk8/hotspot/src/share/vm/interpreter/interpreterRuntime.cpp	Mon Dec 15 09:36:46 2014 +0100
@@ -394,7 +394,11 @@
     // during deoptimization so the interpreter needs to skip it when
     // the frame is popped.
     thread->set_do_not_unlock_if_synchronized(true);
+#ifdef CC_INTERP
+    return (address) -1;
+#else
     return Interpreter::remove_activation_entry();
+#endif
   }
 
   // Need to do this check first since when _do_not_unlock_if_synchronized

